### ISSUE TITLE
linter: rules of a plugin that is not enabled are dropped without a diagnostic

### ISSUE THREAD

**Issue by shnakatani** [oxc-project/oxc#25579]
### What version of Oxlint are you using?

1.76.0

### What command did you run?

`oxlint src/example.ts`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "plugins": ["react"],
  "rules": {
    "typescript/consistent-type-assertions": ["error", { "assertionStyle": "never" }]
  }
}
```

### What happened?

With the config above, `src/example.ts` below exits 0 and reports nothing.

```ts
const value = { a: 1 } as { a: number };
```

Adding `"typescript"` to `plugins` makes the same file fail. The rule entry itself is dropped while the resolved config is built: `oxlint --print-config` shows no `typescript/*` key at all with the config above.

The rule name is still validated in that situation. A typo fails the run even when the plugin is not enabled, so the plugin is resolved before the setting is discarded:

```
$ # with "plugins": ["react"] and "rules": { "vitest/no-such-rule-here": "error" }
Failed to parse oxlint configuration file.
  x Rule 'no-such-rule-here' not found in plugin 'vitest'
```

#### What I measured

| `plugins` | rule written in `rules` | result |
| --- | --- | --- |
| `["react"]` (replaces the default set) | `typescript/consistent-type-assertions` | exit 0, no output |
| omitted (defaults apply) | `vitest/expect-expect` | exit 0, no output |
| `["typescript", "unicorn", "oxc", "react", ...]` | `vitest/expect-expect` | exit 0, no output |
| omitted (defaults apply) | `unicorn/no-null` | exit 1, rule runs |
| any | `vitest/no-such-rule-here` | exit 1, `Rule 'no-such-rule-here' not found in plugin 'vitest'` |
| any | `nonexistent-plugin/some-rule` | exit 1, `Plugin 'nonexistent-plugin' not found` |

Whether the plugin belongs to the default set does not matter. What matters is whether it is enabled at that moment. A default plugin that falls out because `plugins` replaced the set behaves the same as a plugin that was never in the set.

#### Why a diagnostic would help

Our project kept `typescript/consistent-type-assertions` in `rules` while `plugins` was `["react"]`. The rule never ran. Our internal docs said type assertions were caught by lint, and nothing in the lint output contradicted that. We found 9 violations once the plugin was enabled, along with 58 other typescript rule violations.

The three neighbouring cases all fail loudly, which is what made us assume this one would too:

- an unknown plugin name fails
- an unknown rule name fails, including for plugins that are not enabled
- a rule whose plugin is enabled runs

Only the fourth case, a valid rule name under a plugin that is not enabled, is quiet. Since the rule name is already resolved against the plugin, the same place could report that the plugin is not enabled.

#### What I would expect

A warning at config load time, naming the rule and the plugin. Something like:

```
warning: rule 'typescript/consistent-type-assertions' is configured but the 'typescript' plugin is not enabled
```

An error would work too, but it may break configs that carry leftover entries from an ESLint migration, so a warning seems safer.

If the current behaviour is intentional for that migration path, a note in the config file reference would also close the gap for us. The reference explains that `plugins` overwrites the default set, and it does not say what happens to rules whose plugin is left out.

#### Environment

The config lives in `vite.config.ts` under a `lint` block (vite-plus 0.2.8 passes it through as `OxlintConfig`), so the JSON above is the equivalent `.oxlintrc.json`. macOS, aarch64.
